spec/classes/foreman_proxy__plugin__dynflow_spec.rb

```diff
@@ -54,6 +54,13 @@
               ":ssl_private_key: /etc/puppetlabs/puppet/ssl/private_keys/foo.example.com.pem",
             ])
           end
+
+          it 'should restart external core' do
+            should contain_file("#{etc_dir}/smart_proxy_dynflow_core/settings.yml").
+              with_notify('Service[smart_proxy_dynflow_core]')
```

Comment:
Here is a tricky detail: `with_notify` checks that there is a notify property. What you really intended was
```suggestion
              that_notifies('Service[smart_proxy_dynflow_core]')
```

This actually checks the graph and if it exists.